biz.aQute.bnd.reporter/src/biz/aQute/bnd/reporter/plugins/entries/bundle/ChecksumPlugin.java

```diff
@@ -0,0 +1,95 @@
+package biz.aQute.bnd.reporter.plugins.entries.bundle;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import aQute.bnd.annotation.plugin.BndPlugin;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.service.Plugin;
+import aQute.bnd.service.reporter.ReportEntryPlugin;
+import aQute.service.reporter.Reporter;
+import biz.aQute.bnd.reporter.generator.EntryNamesReference;
+import biz.aQute.bnd.reporter.maven.dto.ChecksumDTO;
+
+/**
+ * This plugin calculates and add the checksums of the file to the report.
+ */
+@BndPlugin(name = "entry." + EntryNamesReference.CHECKSUM)
+public class ChecksumPlugin implements ReportEntryPlugin<Jar>, Plugin {
+
+	private Reporter					_reporter;
+	private final Map<String, String>	_properties	= new HashMap<>();
+
+	public ChecksumPlugin() {
+		_properties.put(ReportEntryPlugin.ENTRY_NAME_PROPERTY, EntryNamesReference.CHECKSUM);
+		_properties.put(ReportEntryPlugin.SOURCE_CLASS_PROPERTY, Jar.class.getCanonicalName());
+	}
+
+	@Override
+	public ChecksumDTO extract(final Jar jar, final Locale locale) {
+		Objects.requireNonNull(jar, "jar");
+		Objects.requireNonNull(locale, "locale");
+
+		ChecksumDTO checksumDTO = new ChecksumDTO();
```

Comment:
See the `aQute.libg.cryptography` for classes to compute MD5, SHA1, SHA256. They provide a nice `asHex` method to return the digest value as a hex string.

Reply:
I changed it to `aQute.libg.cryptography`. I is really useful!

Had to fix SHA-512. Not sure with that. please have a look.